src/main/java/com/example/demo/dao/message/MessageDAOImpl.java

```diff
@@ -0,0 +1,52 @@
+package com.example.demo.dao.message;
+
+import com.example.demo.dao.message.MessageDAO;
+import com.example.demo.model.Message;
+import com.example.demo.utils.*;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+import java.util.List;
+
+public class MessageDAOImpl implements MessageDAO {
+    @Override
+    public void save(Message message) {
+        Session session = HibernateSessionFactoryUtil.getSessionFactory().openSession();
```

Comment:
Prefer using https://www.tutorialspoint.com/Automatic-resource-management-in-Java
Current implementation is not exception-secure

Reply:
Session + transaction init may be moved to separated method (I'm wondering if it's not implemented in hibernate)

Reply:
do I understand correctly that the method should return an open session and transaction?

Reply:
As for me, passing lambda action inside looks better